### ISSUE TITLE
SF_NPM_REGISTRY is not working

### ISSUE THREAD

**Issue by gavignon** [forcedotcom/cli#2517]
### Summary
Environment variable SF_NPM_REGISTRY is not working

### Steps To Reproduce
Install a sf plugin hosted on a private npm registry
>- `SF_NPM_REGISTRY=URL sf plugins install {PLUGIN_NAME}`

### Expected result
SF Plugin is installed successfully

### Actual result
Install fails because the private registry is not taken into account

### Workaround
Use SFDX_NPM_REGISTRY variable instead

### System Information
bash and zsh
```json
{
  "cliVersion": "@salesforce/cli/2.11.8",
  "architecture": "darwin-x64",
  "nodeVersion": "node-v16.20.2",
  "osVersion": "Darwin 22.6.0",
  "shell": "zsh",
  "rootPath": "/Users/{USER}/.npm-global/lib/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.8 (core)",
    "@oclif/plugin-commands 2.2.26 (core)",
    "@oclif/plugin-help 5.2.19 (core)",
    "@oclif/plugin-not-found 2.4.1 (core)",
    "@oclif/plugin-plugins 3.8.2 (core)",
    "@oclif/plugin-search 0.0.22 (core)",
    "@oclif/plugin-update 3.2.3 (core)",
    "@oclif/plugin-version 1.3.10 (core)",
    "@oclif/plugin-warn-if-update-available 2.1.0 (core)",
    "@oclif/plugin-which 2.2.32 (core)",
    "@salesforce/cli 2.11.8 (core)",
    "apex 2.3.15 (core)",
    "auth 2.8.17 (core)",
    "data 2.5.12 (core)",
    "deploy-retrieve 1.18.3 (core)",
    "dev 1.1.10 (user)",
    "info 2.6.45 (core)",
    "limits 2.3.35 (core)",
    "login 1.2.31 (core)",
    "marketplace 0.2.2 (core)",
    "org 2.10.8 (core)",
    "packaging 1.20.0 (user)",
    "schema 2.3.28 (core)",
    "settings 1.4.30 (core)",
    "signups 1.4.22 (user)",
    "sobject 0.2.7 (core)",
    "source 2.10.36 (core)",
    "telemetry 2.3.4 (core)",
    "templates 55.5.13 (core)",
    "trust 2.6.15 (core)",
    "user 2.3.32 (core)"
  ]
}
```

**Comment by iowillhoit** [comment]
Thanks @gavignon, It looks like it is setting the registry correctly but our code where we validate signatures only looks for `SFDX_NPM_REGISTRY`. We'll get a fix in!

**Comment by iowillhoit** [comment]
This should be fixed in the `nightly` build. You can try it out at `@salesforce/cli@nightly`. It will be promoted to `latest` on October 25th

**Comment by jshackell-sfdc** [comment]
This issue is fixed in CLI version [2.14.6 (Oct 25, 2023)](https://github.com/forcedotcom/cli/blob/main/releasenotes/README.md#2146-oct-25-2023-stable).
